app/jobs/risc_delivery_job.rb

```diff
@@ -48,12 +48,28 @@ def perform(
         }.to_json,
       )
     end
-  rescue *NETWORK_ERRORS, RedisRateLimiter::LimitError => err
-    raise err if !inline?
+  rescue *NETWORK_ERRORS => err
+    if self.executions < 2
+      raise err if !inline?
+    end
```

Comment:
We can reduce the number of if statements and and simplify the flow slightly I think

```suggestion
    raise err if self.executions < 2 && !inline?
```